sumologic/resource_sumologic_cse_automation.go

```diff
@@ -0,0 +1,147 @@
+package sumologic
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"log"
+)
+
+func resourceSumologicCSEAutomation() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceSumologicCSEAutomationCreate,
+		Read:   resourceSumologicCSEAutomationRead,
+		Delete: resourceSumologicCSEAutomationDelete,
+		Update: resourceSumologicCSEAutomationUpdate,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"playbook_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"cse_resource_type": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"cse_resource_sub_types": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"execution_types": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validation.StringIsNotEmpty,
+				},
+			},
+			"enabled": {
+				Type:     schema.TypeBool,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceSumologicCSEAutomationRead(d *schema.ResourceData, meta interface{}) error {
+	c := meta.(*Client)
+
+	var CSEAutomation *CSEAutomation
+	id := d.Id()
+
+	CSEAutomation, err := c.GetCSEAutomation(id)
+	if err != nil {
+		log.Printf("[WARN] CSE Custom Entity Type not found when looking by id: %s, err: %v", id, err)
+
+	}
+
+	if CSEAutomation == nil {
+		log.Printf("[WARN] CSE Custom Entity Type not found, removing from state: %v - %v", id, err)
+		d.SetId("")
```

Comment:
if I remember correct, and based on the WARN I put in place is to remove from state, in case not found. Meaning the resource was deleted not using terraform ( via UI or API )

Reply:
Oh I see that makes sense, you wouldn't want TF to hold onto it anymore.